tdm-reader/nitdmreader.yml

```diff
@@ -0,0 +1,512 @@
+      parameters:
+        - name: fileIds
+          description: List of File Ids to inspect.
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/FileList'
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/FileInformation'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/files/{file-id}/groups:
+    get:
+      tags: [metadata]
+      summary: Lists all groups in the file
+      operationId: getChannelPropertiesCount
+      produces:
+        - application/json
+      parameters:
+        - name: file-id
+          description: File id
+          in: path
+          required: true
+          type: string
+        - name: skip
+          description: Number of items to skip
+          in: query
+          type: integer
+          default: 0
+        - name: take
+          description: Number of items to take after skipping
+          in: query
+          type: integer
+          default: -1
+        - name: returnCount
+          description: Specify this to return the total count of items
+          in: query
+          type: boolean
+          default: false
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/NamesList'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/files/{file-id}/groups/{group-name}/channels:
+    get:
+      tags: [metadata]
+      summary: Lists all channels in the group
+      operationId: getChannels
+      produces:
+        - application/json
+      parameters:
+        - name: file-id
+          description: File id
+          in: path
+          required: true
+          type: string
+        - name: group-name
+          description: Group name
+          in: path
+          required: true
+          type: string
+        - name: skip
+          description: Number of items to skip
+          in: query
+          type: integer
+          default: 0
+        - name: take
+          description: Number of items to take after skipping
+          in: query
+          type: integer
+          default: -1
+        - name: returnCount
+          description: Specify this to return the total count of items
+          in: query
+          type: boolean
+          default: false
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/NamesList'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/Error'
+  /v1/query-data:
+    post:
+      tags: [data]
+      summary: Returns data from channels.
+      operationId: queryData
+      produces:
+        - application/json
+      parameters:
+        - name: ChannelSpecificationsAndWindow
+          description: Channel specifications and window information.
```

Comment:
Do we expect users to know what window information is? Just something to think about